### ISSUE TITLE
Root Schema conflicting/merging with local <JsonForms /> Schema

### ISSUE THREAD

**Issue by wootencl** [eclipsesource/jsonforms#923]
First off, thanks for the excellent package(s)! Really like the architecture/approach you all are taking to a JSON schema form renderer. 

Was curious if you all might be able to provide an example of JSONForms inferring a UI schema. Currently when I omit it (as `undefined`) or pass in an empty object I get a `No applicable renderer found.` being displayed. Happy to fork the seed app and provide an example if needed.

**Comment by eneufeld** [comment]
Hi, 
could you provide your schema? We already have an example for inferring an UISchema: http://jsonforms.io/examples/generate-uischema .
Without seeing the schema I would assume, that it contains elements which the UI Generator cannot handle, eg an object.

Cheers

**Comment by wootencl** [comment]
Hi @eneufeld. Thanks for the super prompt reply! Sorry it took me a bit to respond. Is the code that backs that example publicly available somewhere?

As for my schema, I'm messing with the seed [app](https://github.com/eclipsesource/jsonforms-react-seed) and was thinking I was having trouble with the ui inference. After some more tinkering that doesn't seem to be the case (will change the title of this issue to better reflect my question). In essence I was thinking you might be able to initialize with nothing and handle it on a form by form basis (i.e. - call `Actions.init` with no/undefined params). Looks like I can omit all but the second parameter (schema) and it will still works. What I can't figure out though is that it seems to use the `schema` specified at `init` in combination with the `schema` specified directly on the `<JsonForms />` element.

Hopefully that makes sense 😬. Happy to provide an example if not.

**Comment by edgarmueller** [comment]
@wootencl I'm not sure if I understand your question correctly, so let me recap it in a different way: 

When specifying the `schema`, but not the `uischema`, on the `JsonForms` component, it uses the derived UI schema, which relates to the one derived from the `init` action creator instead of the specified `schema`.

Is that correct? It would make sense, since the [derivation of the UI schema happens via a default argument in `init`](https://github.com/eclipsesource/jsonforms/blob/master/packages/core/src/actions/index.ts#L46). 

If that's the case I think you could generate the UI schema yourself as a workaround:

```jsx
import { Generate } from '@jsonforms/core';

<JsonForms schema={schema} uischema={Generate.uiSchema(schema)} />
```

Let us know whether this is what you meant.

**Comment by wootencl** [comment]
Sounds like exactly the problem I'm having (`uiSchema` being derived from the `store` instead of the specified `schema`). I'll give that `Generate` method a go to see if it resolves my issue.

Related, it seems like if there's a `schema` specified on the form (indicating non usage of the store's `schema`) and no `uiSchema` then the form would try to generate `uiSchema` based off of that instead (i.e. - making the path you mentioned above default behavior). If so, this could potentially be a feature request.

**Comment by edgarmueller** [comment]
Yes, that makes sense, we'll schedule it for the next milestone. Thanks for the report.

**Comment by wootencl** [comment]
No problem 🙂. Thanks for the quick support/turnaround. Keep up the great work!

**Comment by edgarmueller** [comment]
Fixed via #965, will be released with 2.0.1
